toy_settings/domain/projections.py

```diff
@@ -1,37 +1,46 @@
 from __future__ import annotations
 
+from collections import defaultdict
 from functools import singledispatch
 from typing import Iterable
 
+import attrs
+
 from . import events
 
 
-def current_settings(history: Iterable[events.Event]) -> dict[str, str]:
-    settings: dict[str, str] = {}
+@attrs.define
+class Setting:
+    value: str | None = None
+    idx: int = -1
```

Comment:
Question: Why `-1`?

Reply:
it can be incremented for new settings, None cannot